### ISSUE TITLE
Ubuntu installer "package is of bad quality"

### ISSUE THREAD

**Issue by perry-mitchell** [buttercup-pw/buttercup#25]
When installing on Ubuntu, the software manager shows the notice "package is of bad quality".

**Comment by snhasani** [comment]
and in linux mint x64 v17.3

```bash
(Reading database ... 167145 files and directories currently installed.)
Preparing to unpack buttercup-0.1.0-x64.deb ...
Unpacking buttercup (0.1.0) ...
dpkg-deb (subprocess): cannot copy archive member from 'buttercup-0.1.0-x64.deb' to decompressor pipe: unexpected end of file or stream
dpkg-deb (subprocess): decompressing archive member: lzma error: unexpected end of input
dpkg-deb: error: subprocess <decompress> returned error exit status 2
dpkg: error processing archive buttercup-0.1.0-x64.deb (--install):
 cannot copy extracted data for './usr/share/applications/buttercup/Buttercup' to '/usr/share/applications/buttercup/Buttercup.dpkg-new': unexpected end of file or stream
Processing triggers for gnome-menus (3.10.1-0ubuntu2) ...
Processing triggers for desktop-file-utils (0.22-1ubuntu1) ...
Processing triggers for bamfdaemon (0.5.1+14.04.20140409-0ubuntu1) ...
Rebuilding /usr/share/applications/bamf-2.index...
Processing triggers for mime-support (3.54ubuntu1.1) ...

Processing triggers for mime-support (3.54ubuntu1.1) ...
Errors were encountered while processing:
 buttercup-0.1.0-x64.deb
```

**Comment by perry-mitchell** [comment]
@snhasani Hmm.. that's interesting. I haven't actually installed the deb from the command-line yet, so I haven't seen any messages like that.

Are you sure that's related to the "bad quality" of the package, and not just an error?

**Comment by cobbaroo1** [comment]
Can confirm on Ubuntu 15.10. Full error text attached.
[ErrorText.txt](https://github.com/buttercup-pw/buttercup/files/121081/ErrorText.txt)

**Comment by perry-mitchell** [comment]
@cobbaroo1 @snhasani Are you installing by command line? I'm able to install fine in the ubuntu software manager, even though it alerts that it's a "bad quality" package.

@cobbaroo1 Judging by your error.txt (thanks!), there's a myriad of problems, mostly permissions. I don't think we can solve all of them, which is a pain. Maybe someone knows a way around this?

**Comment by cobbaroo1** [comment]
@perry-mitchell I was using the software center. I was scared by the error and decided to report here.

Possible solution: Find out what permissions are needed to be valid. Create a script to fix permissions to valid, add, commit and push changes, then restore permissions to 700 or whatever you need.

**Comment by perry-mitchell** [comment]
Provided by @MHM5000 : https://gist.github.com/MHM5000/3b13bcd855061d5ccde13fc9892df1e5

**Comment by perry-mitchell** [comment]
Has this been addressed now? I'll probably re-test tomorrow.

**Comment by sallar** [comment]
@perry-mitchell It should be... but requires testing on a real linux machine, so you can maybe have a look at that... Thanks!

**Comment by perry-mitchell** [comment]
Closing as no longer an issue on Ubuntu 16.04. If someone reproduces this again on the new build we can re-open.

**Comment by meowingtwurtle** [comment]
Can confirm resolved on release v0.3.0
